core/node/rules/can_add_event.go

```diff
@@ -198,8 +204,22 @@ func (params *aeParams) canAddChannelPayload(payload *StreamEvent_ChannelPayload
 			requireChainAuth(params.channelMessageWriteEntitlements).
 			onChainAuthFailure(params.onEntitlementFailureForUserEvent)
 	case *ChannelPayload_Redaction_:
-		return aeBuilder().check(params.creatorIsMember).
+		return aeBuilder().
+			check(params.creatorIsMember).
 			requireChainAuth(params.redactChannelMessageEntitlements)
+	case *ChannelPayload_Pin_:
+		pinRuls := &aeChannelPinRules{
+			params: params,
+			pin:    content.Pin,
+		}
+		return aeBuilder().
+			check(params.creatorIsMember).
+			check(pinRuls.validChannelPin).
+			requireChainAuth(params.channelMessageWriteEntitlements)
+	case *ChannelPayload_Unpin_:
+		return aeBuilder().
+			check(params.creatorIsMember).
```

Comment:
@texuf did you want to check that the message is currently pinned? So we don't accept multiple unpin events for the same message, or for a non-existent pin

Reply:
It’s probably a good idea. If you want to add it go ahead.